chimney/src/main/scala/io/scalaland/chimney/internal/compiletime/Configurations.scala

```diff
@@ -0,0 +1,149 @@
+package io.scalaland.chimney.internal.compiletime
+
+import io.scalaland.chimney.dsl.ImplicitTransformerPreference
+import io.scalaland.chimney.internal
+import io.scalaland.chimney.partial
+import io.scalaland.chimney.dsl as dsls
+
+import scala.annotation.nowarn
+
+@nowarn("msg=The outer reference in this type test cannot be checked at run time.")
```

Comment:
Any idea why it is raised? Maybe add a TODO comment so that we don't forget to go back to it before release?

Reply:
For Scala compiler _every_ `trait`, `abstract class` and `case class` which is not `private` is treated as potentially leaking, and accessible from e.g. other instances of our macro class. As such type tests would not catch that this is e.g. `macro2.DerivationResult` and not `macro1.DerivationResult` and compiler complains about it.

I tried using `protected[this]` in attempt to make it visible only to current instance but:
* it didn't restrict access at all, warning remained
* `protected[this]` is deprecated in Scala :P